### PR TITLE
handle optional status code argument in response redirect

### PR DESCRIPTION
**Pull request by cwebley**

### CODE REVIEW

File: lib/response.js [hunk 1]
```diff
@@ -40,6 +40,14 @@ Response.prototype.redirect = function (u) {
 	// In a timeout because page.js did it, but I believe it is better
 	// to let the current route handler finish its stuff before starting
 	// another round of routing, otherwise we could get some weird behavior.
+
+	// Discard the status code argument if it exists
+	if (arguments.length === 2) {
```
Comment (wesleytodd) [inline lib/response.js]: Would rather not use arguments for this.  Just define both arguments and type check:

```javascript
Response.prototype.redirect = function (status, url) {
  if (typeof status === 'string') {
    url = status;
    status = 302;
  }
  // ...
};
```

**Comment by wesleytodd** [comment]
When you make that change, make sure to `commit --ammend` so that this is just one commit.  I want to actually cherry pick this back to 1.x as well as merge to 2.x

**Comment by cwebley** [comment]
wellllll that didn't work. going to rebase

File: lib/response.js [hunk 2]
```diff
@@ -34,12 +34,19 @@ inherits(Response, EventEmitter);
...
  */
-Response.prototype.redirect = function (u) {
+Response.prototype.redirect = function (statusCode, u) {
 	// In a timeout because page.js did it, but I believe it is better
 	// to let the current route handler finish its stuff before starting
 	// another round of routing, otherwise we could get some weird behavior.
+
+	// Sort out arguments by handling the optional statusCode
+	if (typeof statusCode === 'string') {
+		u = statusCode;
```
Comment (wesleytodd) [inline lib/response.js]: You would also need to reset status code, because here it will still be the url.

**Comment by wesleytodd** [comment]
Did you `git commit --amend && git push -f`?

File: lib/response.js (hunk 2 above)
Comment (cwebley) [reply lib/response.js]: ohh didn't realize we're even using status at all on the front end. figured we're just discarding it. i see it though

**Comment by cwebley** [comment]
didn't git push -f, thats the problem

File: lib/response.js (hunk 2 above)
Comment (wesleytodd) [reply lib/response.js]: We aren't, but I figure it would clear up any possible issues in the future if we find a use for it

**Comment by wesleytodd** [comment]
Ok, this is published in both `2.0.0-BETA-3` and `1.5.0`.

**Pull request merged**